src/local/base_node.rs

```diff
@@ -453,7 +453,242 @@ mod tests {
+                let actual_result = node_ref.search_by_id(&req).unwrap();
+
+                let expected_result = lt_clone
+                    .left_neighbors()
+                    .unwrap()
+                    .into_iter()
+                    .filter(|(l, id)| *l <= req.level() && id.id() >= req.target())
+                    .min_by_key(|(_, id)| *id.id());
+
+                match expected_result {
+                    Some((expected_lvl, expected_identity)) => {
+                        assert_eq!(expected_lvl, actual_result.level());
+                        assert_eq!(*expected_identity.id(), *actual_result.result());
+                    }
+                    None => {
+                        // If no expected result, it should return its own identifier
+                        assert_eq!(actual_result.level(), 0);
+                        assert_eq!(*actual_result.result(), *node_ref.get_identifier());
+                    }
+                }
+            });
+            handles.push(handle);
+        }
+
+        // Ensures all threads are ready to run before the main thread tries to join them
+        // avoiding a situation where the main thread tries to join thread that haven't started yet.
+        barrier.wait();
+        let timeout = std::time::Duration::from_millis(1000);
+        join_all_with_timeout(handles.into_boxed_slice(), timeout).unwrap();
+    }
+
+    /// Tests the `search_by_id` method of a `LocalNode` under concurrent conditions where multiple
+    /// threads perform searches in the right direction (`Direction::Right`) simultaneously.
+    ///
+    /// The test:
+    /// - Creates a `LocalNode` with a random identifier, random membership vector,
+    ///   and a lookup table (`RandomLookupTable`).
+    /// - Randomly generates a target identifier to search for.
+    /// - Spawns 20 threads that conduct searches concurrently from the node.
+    ///
+    /// ### Test Specific Logic:
+    /// - Each thread constructs a search request targeting the same identifier and executes the `search_by_id` method.
+    /// - The expected search result is derived by finding the closest matching identifier from the
+    ///   right neighbors in the lookup table (`lt`) that meets the search criteria (e.g., level,
+    ///   target identifier comparison).
+    /// - If no valid neighbor is found, it expects the result to default to the `LocalNode`'s own identifier.
+    ///
+    /// ### Assertions:
+    /// - If a valid neighbor exists, the search output should match both the level and identifier.
+    /// - If no valid neighbor exists, the search result should match the node's own identifier at level 0.
+    #[test]
+    fn test_search_by_id_concurrent_right_direction() {
+        let lt = random_lookup_table_with_extremes(LOOKUP_TABLE_LEVELS);
+        let target = random_identifier();
+
+        let node = Arc::new(LocalNode {
+            id: random_identifier(),
```

Comment:
Echoing my previous comment here: To ensure that the test scenario does not trigger the quality case, that is, the target is expected to be among the right neighbors of the node, if any, but not the node itself, it is prudent to check that the node’s id is not equal to the target's.